### DISCUSSION TITLE
How to automatically delete rows from table after a specified duration in a ticket booking system

### DISCUSSION THREAD

**Discussion by lordvcs** [pocketbase/pocketbase#4610] [Q&A] [answered]
I am building an application like a movie ticket booking system using pocketbase, where the user can select their seats and proceed to checkout. After selecting the seats, the user has 10 mins to complete the payment to confirm the booking and during that 10 mins, the selected seats become unavailable to others for booking. To enable this functionality I create a new row in the `orders` table with payment status as `pending`, which also keeps track of the tickets selected. 
Now the question is, how can I automatically delete a row from `orders` table where payment status is pending, when the `created` time is more than 10 mins ago?.

I think its possible to do this using a cronjob, but I would prefer a solution without cron. But if there are no other solutions, a cronjob solution will be fine too.

BTW, thanks for everyones contributin to this project, I love it.

**Comment by ganigeorgiev** [comment, marked as answer]
There is no other option at the moment other than a scheduling a cleanup task. For jobs scheduling you can refer to:

- Go: https://pocketbase.io/docs/go-jobs-scheduling/
- JS `pb_hooks`: https://pocketbase.io/docs/js-jobs-scheduling/
